packages/next/build/entries.ts

```diff
@@ -319,9 +319,7 @@ export async function createEntrypoints(params: CreateEntrypointsParams) {
         !absolutePagePath.startsWith(ROOT_DIR_ALIAS) &&
         /[\\\\/]_middleware$/.test(page)
       ) {
-        throw new Error(
-          `nested Middleware is not allowed (found pages${page}) - https://nextjs.org/docs/messages/nested-middleware`
-        )
+        throw new NestedMiddlewareError(page, rootDir, pagesDir)
```

Comment:
Is this only reporting the first instance of nested middleware? It would be nice if we reported all of them so the user is able to update them all at once instead of fixing one and then seeing the error again to fix another

Reply:
All the nested middleware files are caught, although in different error messages:

<img width="599" alt="image" src="https://user-images.githubusercontent.com/186268/172565865-b4b802eb-0eca-4788-b4bf-5c5be2b0dea2.png">

(please note that in the example above, I'm using `/src/pages`)

Collecting all of them in the same error message may be tricky, because we would need to wait for Webpack to complete its analysis, and then display the error.
Do you think it worth it?

Reply:
Couldn't we catch them all since we have all of the entries during build and then show the error once and then during dev we already catch them all they are just logged separately but we could log them together if we waited until after iterating over the paths [here](https://github.com/vercel/next.js/blob/977ff5247222acfdf0b7429fa08c9f95adcb030e/packages/next/server/dev/next-dev-server.ts#L355)

Reply:
Thanks for the tip, I'll get it done!

Reply:
Went ahead and updated this real quick in https://github.com/vercel/next.js/pull/37513/commits/e7a2dc00d287bb824a43e7c25932d6aed6e31d4f so we can get this in the next canary